src/main/java/com/example/cdr/eventsmanagementsystem/Controller/AdminController/AdminUserController.java

```
@@ -30,19 +30,76 @@ public record CreateUserRequest(
     if (req.password() == null || req.password().length() < 8) return ResponseEntity.badRequest().body("password must be at least 8 chars");
     if (req.email() == null || req.email().isBlank()) return ResponseEntity.badRequest().body("email is required");
 
-    String id = keycloakService.createUser(
-        req.username().trim(), req.email().trim(), req.firstName(), req.lastName(), req.password(), req.role()
+    String id = keycloakService.createUserByAdmin(
+        req.username().trim(),
+        req.email().trim(),
+        req.firstName(),
+        req.lastName(),
+        req.role(),
+        req.password() // only for record/UI; login requires the first-login password change
+    );
+    return ResponseEntity.ok(Map.of("id", id));
+  }
+
+  // ------------------------ SELF REGISTER (PERMANENT PW) ------------------------
+  @Operation(
+      summary = "Register user (self/public flow)",
+      description = """
+        Creates a user with a PERMANENT password (no first-login screen) and assigns a default role \
+        (e.g., 'attendee'). Useful if you expose a public signup backed by your API instead of Keycloak's \
+        built-in registration. Returns the created user id.
+        """
+  )
+  @PostMapping(AdminControllerConstants.ADMIN_USERS_SELF_REGISTER_URL)
+  public ResponseEntity<?> selfRegister(@RequestBody SelfRegisterRequest req) {
+    if (req.username() == null || req.username().isBlank()) return ResponseEntity.badRequest().body("username is required");
+    if (req.password() == null || req.password().length() < 8) return ResponseEntity.badRequest().body("password must be at least 8 chars");
+    if (req.email() == null || req.email().isBlank()) return ResponseEntity.badRequest().body("email is required");
+
+    String role = (req.defaultRole() == null || req.defaultRole().isBlank()) ? "attendee" : req.defaultRole().trim();
```

I believe we can throw an exception if no role is provided instead of defaulting to `attendee`, since that was the main cause of issue #107.

---

okay sure